nexus/src/external_api/http_entrypoints.rs

```diff
@@ -1558,7 +1682,10 @@ async fn project_vpcs_put_vpc(
 }
 
 /**
- * Delete a vpc from a project.
+ * Delete a VPC
+ *
+ * Permanently delete a VPC from a project. This operation cannot be undone. Any subnets
+ * or other resources in the VPC will be deleted as well.
```

Comment:
I'm assuming, for the sake of brevity, these comments are meant to briefly describe the common, successful case and not go into all the gory semantic details of a call? E.g. I would hope that if the VPC contains in-use subnets/interfaces that we return a result indicating the operation cannot be performed.

Reply:
How about as a compromise there is a parameter `force` such that if `?force=true` it cleans up all the child resources. This would help the console as well as prevent someone from fucking something up accidentally. And in the case the person actually wants to delete everything they can change a param and boom versus writing a bunch more code for dependencies which is always super annoying. And the console shows you everything you will delete in deleting a resource and takes the effort off them to do the child calls as well. cc @zephraph to verify console use case

Reply:
It's not just child resources that would need cleaning. E.g., this VPC might include a subnet with N interfaces. Those N interfaces are used by N running guests. Those interfaces cannot go away while the guests are running. So not only would we have to cleanup all child resources, we have to also find, stop, and delete all other resources using these child resources. I mentions guests here because it's the obvious thing that comes to mind, but perhaps there are other non-child resources that would need to be dealth with as well. So should deleting a VPC cause stopping/deletion of other things that are not under the VPC umbrella, but are instead consumers of these resources?

But that aside, in the case of not passing that flag we still have an error to deal with. My larger question was if we intend to document that stuff here or if these comments are only for the obvious working case, and we'll elaborate all the gritty details in some other place?

Reply:
Ah for sure can elaborate on the instances in that case I would even expect `force` to fail.

> My larger question was if we intend to document that stuff here or if these comments are only for the obvious working case, and we'll elaborate all the gritty details in some other place?
 
For consumers of the API, these comments are being populated by the docs, so it would be really nice to get all details here. since it will show up here: https://docs.oxide.computer/api/delete-a-vpc-from-a-project as they are implementing stuff. Also the way I wrote all the clients these comments propagate to the clients so if they use godoc or rust docs its right there in the code as well. Best to be as verbose as possible. We will obviously have other tutorials but being as verbose as possible here would be really nice!

Reply:
Maybe we dont even need force in this case, basically I'm trying to account for a VPC with things in it that arent being used and someone just wants to clean up. Like for instance for terraform etc, would be nice to do one call versus n+1

Reply:
That being said, these docs are just a first pass it doesn't need to be ship-worthy but also when will we have time to do ship worthy things haha another question

Reply:
In the case of a failure to delete by in-use resources, we really just want to make sure we can communicate what needs to be cleaned up or removed from use before the desired user action can happen.